datasync/chaindatafetcher/chaindata_fetcher.go

```diff
@@ -233,53 +250,56 @@ func (f *ChainDataFetcher) SetComponents(components []interface{}) {
-	f.wg.Add(1)
-	defer f.wg.Done()
-	for {
-		select {
-		case <-f.stopCh:
-			logger.Info("handleRequest is stopped")
+func (f *ChainDataFetcher) handleChainEvent(reqType requestType, ev blockchain.ChainEvent) {
+	// TODO-ChainDataFetcher parallelize handling data
+	if checkRequestType(reqType, requestTypeTransaction) {
+		if err := retryFunc(f.repo.InsertTransactions)(ev); err != nil {
+			logger.Error("failed to insert transactions", "err", err, "reqType", reqType, "blockNumber", ev.Block.NumberU64())
 			return
-		case req := <-f.reqCh:
-			// TODO-ChainDataFetcher parallelize handling data
-			if checkRequestType(req.reqType, requestTypeTransaction) {
-				if err := f.repo.InsertTransactions(req.event); err != nil {
-					f.resCh <- newResponse(req.reqType, req.event.Block.Number(), err)
-				}
-			}
+		}
+	}
 
-			if checkRequestType(req.reqType, requestTypeTokenTransfer) {
-				if err := f.repo.InsertTokenTransfers(req.event); err != nil {
-					f.resCh <- newResponse(req.reqType, req.event.Block.Number(), err)
-				}
-			}
+	if checkRequestType(reqType, requestTypeTokenTransfer) {
+		if err := retryFunc(f.repo.InsertTokenTransfers)(ev); err != nil {
+			logger.Error("failed to insert token transfers", "err", err, "reqType", reqType, "blockNumber", ev.Block.NumberU64())
+			return
+		}
+	}
 
-			if checkRequestType(req.reqType, requestTypeContracts) {
-				if err := f.repo.InsertContracts(req.event); err != nil {
-					f.resCh <- newResponse(req.reqType, req.event.Block.Number(), err)
-				}
-			}
+	if checkRequestType(reqType, requestTypeContracts) {
+		if err := retryFunc(f.repo.InsertContracts)(ev); err != nil {
+			logger.Error("failed to insert contracts", "err", err, "reqType", reqType, "blockNumber", ev.Block.NumberU64())
+			return
+		}
+	}
 
-			if checkRequestType(req.reqType, requestTypeTraces) {
-				if err := f.repo.InsertTraceResults(req.event); err != nil {
-					f.resCh <- newResponse(req.reqType, req.event.Block.Number(), err)
-				}
-			}
+	if checkRequestType(reqType, requestTypeTraces) {
+		if err := retryFunc(f.repo.InsertTraceResults)(ev); err != nil {
+			logger.Error("failed to insert trace results", "err", err, "reqType", reqType, "blockNumber", ev.Block.NumberU64())
+			return
 		}
 	}
+	f.resCh <- newResponse(reqType, ev.Block.Number(), nil)
 }
 
-func (f *ChainDataFetcher) reqLoop() {
+func (f *ChainDataFetcher) handleRequest() {
 	f.wg.Add(1)
 	defer f.wg.Done()
 	for {
 		select {
 		case <-f.stopCh:
-			logger.Info("stopped reqLoop for chaindatafetcher")
+			logger.Info("handleRequest is stopped")
 			return
 		case ev := <-f.chainCh:
-			f.reqCh <- newRequest(requestTypeAll, ev)
+			f.handleChainEvent(requestTypeAll, ev)
+		case req := <-f.reqCh:
+			ev, err := f.makeChainEvent(req.block)
+			if err != nil {
+				// TODO-ChainDataFetcher handle error
+				logger.Error("making chain event is failed", "err", err)
+				break
+			}
+			f.handleChainEvent(req.reqType, ev)
```

Comment:
For me, the relation between chain event and request seems reversed. 
Making chain event from the request does not seem natural, as the request is not a chain event.

Reply:
Renamed `handleChainEvent` to `handleRequestByType`. bec68e063cb9fdf9ffcc38e7903ed99a35ff1657